shared/message/vision/Ball.proto

```diff
@@ -39,6 +39,7 @@ message Ball {
         fmat3 covariance = 3;
     }
     /// Cone describing the ball from the camera's perspective
+    /// See Geometry.proto
```

Comment:
Rather than using a cone object, I think it might be better to just have an axis and radius directly in here. The original idea was to have a common cone object so we could use it for math functions that worked with cones. But that never happened so now we just have badly named variables for cones

Reply:
While we're here, why is there a repeated Measurement? It seems to only add in one Measurement in BallDetector, seems it shouldn't be repeated and the Measurement components could be pulled out into here. Type doesn't seem to be used either.

Reply:
It is repeated because localisation should, theoretically, handle multiple measurements for each object. The idea is that you measure the distance to the ball using multiple, distinct, methods to help improve the localisation accuracy. We never ended up implementing any other measurement methods for the ball, but left in the ability to handle multiple measurements in case someone ever came up with one.

This multiple measurement thing should be done for all vision objects (balls, goals, field lines, center circle, obstacles_ wherever possible. @TrentHouliston might be able to explain this better

Reply:
If it's something we should, do, then I'll see about updating this to use both projection and angular based methods? And revert removing it

Reply:
Im pretty sure that projection isn't distinct from angular (I think the two measurements are derived from each other or something?). @TrentHouliston knows more about this than I do, but there was a reason we didnt add that measurement

Reply:
Projection-based and width based are distinct, I don't know why they were removed. My guess is someone was annoyed that the projection-based was bad because odometry was bad.